internal/service/advancedcluster/model_advanced_cluster.go

```diff
@@ -644,6 +663,9 @@ func flattenAdvancedReplicationSpecRegionConfigSpec(apiObject *admin20231115.Ded
 				tfMap["ebs_volume_type"] = apiObject.GetEbsVolumeType()
 			}
 		}
+		if _, ok := tfMapObject["disk_size_gb"]; ok {
+			tfMap["disk_size_gb"] = apiObject.GetDiskSizeGB()
+		}
```

Comment:
do we need to check if the attribute is defined in the config? Since it is compute I see not issue in storing the result in the state always

Reply:
I have maintained the previous logic here so we should be able to use this for the resource as well